lib/lsblk/json.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 Resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const getMountpoints = (children) => {
+  return children.filter((child) => {
+    return child.mountpoint;
+  }).map((child) => {
+    return {
+      path: child.mountpoint,
+      label: child.label || child.partlabel
+    };
+  });
+};
+
+const getDescription = (device) => {
+  const description = [
+    device.label || '',
+    device.vendor || '',
+    device.model || ''
+  ];
+  if (device.children) {
+    let subLabels = device.children
+      .filter((c) => {
+        return c.label && c.label !== device.label;
+      })
+      .map((c) => {
+        return c.label;
+      });
+    subLabels = Array.from(new Set(subLabels));
+    if (subLabels.length) {
+      description.push(`(${subLabels.join(', ')})`);
+    }
+  }
+  return description.join(' ').replace(/\s+/g, ' ').trim();
+};
+
+const transform = (data) => {
+  return data.blockdevices.map((device) => {
+    const isVirtual = /^(block)$/i.test(device.subsystems);
+    const isSCSI = /^(sata|scsi|ata|ide|pci)$/i.test(device.tran);
+    const isUSB = /^(usb)$/i.test(device.tran);
+    const isReadOnly = Number(device.ro) === 1;
+    const isRemovable = Number(device.rm) === 1 || isVirtual;
+    return {
+      enumerator: 'lsblk',
+      busType: (device.tran || 'UNKNOWN').toUpperCase(),
+      busVersion: null,
+      device: device.name,
+      raw: device.kname,
+      description: getDescription(device),
+      error: null,
+      size: Number(device.size),
```

Comment:
What happens if these evaluate to `NaN`? Should we handle this somehow and make them undefined or something like that?

Reply:
Defaulted to `null` now, although that shouldn't happen.